### ISSUE TITLE
IJ 2018.1 ClassNotFoundException: org.jetbrains.kotlin.script.ScriptTemplatesProvider 

### ISSUE THREAD

**Issue by krasa** [dkandalov/live-plugin#79]
Happens on the start, not sure if the problem is in this plugin.

```
org/jetbrains/kotlin/script/ScriptTemplatesProvider
com.intellij.openapi.extensions.impl.PicoPluginExtensionInitializationException: org/jetbrains/kotlin/script/ScriptTemplatesProvider
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getComponentInstance(ExtensionComponentAdapter.java:96)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getExtension(ExtensionComponentAdapter.java:119)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:241)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensions(ExtensionPointImpl.java:190)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.getContributors(ScriptDefinitionsManager.kt:80)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.reloadScriptDefinitions(ScriptDefinitionsManager.kt:87)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.currentDefinitions(ScriptDefinitionsManager.kt:64)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.findScriptDefinition(ScriptDefinitionsManager.kt:71)
	at org.jetbrains.kotlin.script.ScriptDefinitionProvider$DefaultImpls.findScriptDefinition(KotlinScriptDefinitionProvider.kt:27)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.findScriptDefinition(ScriptDefinitionsManager.kt:44)
	at org.jetbrains.kotlin.script.KotlinScriptDefinitionProviderKt.getScriptDefinition(KotlinScriptDefinitionProvider.kt:36)
	at org.jetbrains.kotlin.idea.core.script.dependencies.KotlinScriptResolveScopeProvider.getResolveScope(KotlinScriptResolveScopeProvider.kt:31)
	at com.intellij.psi.impl.file.impl.ResolveScopeManagerImpl.lambda$new$0(ResolveScopeManagerImpl.java:63)
	at com.intellij.util.containers.ConcurrentFactoryMap$5.create(ConcurrentFactoryMap.java:234)
	at com.intellij.util.containers.ConcurrentFactoryMap.get(ConcurrentFactoryMap.java:62)
	at com.intellij.psi.impl.file.impl.ResolveScopeManagerImpl.getResolveScopeFromProviders(ResolveScopeManagerImpl.java:81)
	at com.intellij.psi.impl.file.impl.ResolveScopeManagerImpl.getResolveScope(ResolveScopeManagerImpl.java:143)
	at com.intellij.psi.impl.ResolveScopeManager.getElementResolveScope(ResolveScopeManager.java:51)
	at com.intellij.psi.impl.PsiElementBase.getResolveScope(PsiElementBase.java:184)
	at com.intellij.psi.impl.ElementPresentationUtil.getClassKindImpl(ElementPresentationUtil.java:131)
	at com.intellij.psi.impl.ElementPresentationUtil.lambda$getClassKind$0(ElementPresentationUtil.java:106)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:60)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.ElementPresentationUtil.getClassKind(ElementPresentationUtil.java:109)
	at com.intellij.psi.impl.ElementPresentationUtil.getFlags(ElementPresentationUtil.java:52)
	at com.intellij.psi.impl.ElementPresentationUtil.createLayeredIcon(ElementPresentationUtil.java:64)
	at com.intellij.psi.impl.PsiClassImplUtil.lambda$static$0(PsiClassImplUtil.java:261)
	at com.intellij.ui.IconDeferrerImpl.deferImpl(IconDeferrerImpl.java:78)
	at com.intellij.ui.IconDeferrerImpl.defer(IconDeferrerImpl.java:68)
	at com.intellij.psi.impl.PsiClassImplUtil.getClassIcon(PsiClassImplUtil.java:281)
	at com.intellij.psi.impl.PsiClassImplUtil.getClassIcon(PsiClassImplUtil.java:268)
	at com.intellij.psi.impl.source.PsiClassImpl.getElementIcon(PsiClassImpl.java:633)
	at com.intellij.psi.impl.ElementBase.computeIconNow(ElementBase.java:119)
	at com.intellij.psi.impl.ElementBase.lambda$static$0(ElementBase.java:59)
	at com.intellij.ui.DeferredIconImpl.evaluate(DeferredIconImpl.java:278)
	at com.intellij.ui.DeferredIconImpl.lambda$null$0(DeferredIconImpl.java:163)
	at com.intellij.ui.IconDeferrerImpl.evaluateDeferred(IconDeferrerImpl.java:111)
	at com.intellij.ui.DeferredIconImpl.lambda$null$1(DeferredIconImpl.java:163)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1142)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:84)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$2(ProgressIndicatorUtils.java:140)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:580)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:525)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:85)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:138)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:84)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:105)
	at com.intellij.ui.DeferredIconImpl.lambda$paintIcon$4(DeferredIconImpl.java:162)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NoClassDefFoundError: org/jetbrains/kotlin/script/ScriptTemplatesProvider
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
	at com.intellij.util.lang.UrlClassLoader._defineClass(UrlClassLoader.java:289)
	at com.intellij.util.lang.UrlClassLoader.defineClass(UrlClassLoader.java:285)
	at com.intellij.util.lang.UrlClassLoader._findClass(UrlClassLoader.java:254)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClassInsideSelf(PluginClassLoader.java:142)
	at com.intellij.ide.plugins.cl.PluginClassLoader.tryLoadingClass(PluginClassLoader.java:74)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:61)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.loadImplementationClass(ExtensionComponentAdapter.java:158)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getComponentImplementation(ExtensionComponentAdapter.java:66)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getComponentInstance(ExtensionComponentAdapter.java:73)
	... 55 more
Caused by: java.lang.ClassNotFoundException: org.jetbrains.kotlin.script.ScriptTemplatesProvider PluginClassLoader[LivePlugin, 0.6.1 beta] com.intellij.ide.plugins.cl.PluginClassLoader@7563e062
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:63)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 70 more
Caused by:

java.lang.NoClassDefFoundError: org/jetbrains/kotlin/script/ScriptTemplatesProvider
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
	at java.lang.ClassLoader.defineClass(ClassLoader.java:642)
	at com.intellij.util.lang.UrlClassLoader._defineClass(UrlClassLoader.java:289)
	at com.intellij.util.lang.UrlClassLoader.defineClass(UrlClassLoader.java:285)
	at com.intellij.util.lang.UrlClassLoader._findClass(UrlClassLoader.java:254)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClassInsideSelf(PluginClassLoader.java:142)
	at com.intellij.ide.plugins.cl.PluginClassLoader.tryLoadingClass(PluginClassLoader.java:74)
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:61)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at java.lang.Class.forName0(Native Method)
	at java.lang.Class.forName(Class.java:348)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.loadImplementationClass(ExtensionComponentAdapter.java:158)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getComponentImplementation(ExtensionComponentAdapter.java:66)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getComponentInstance(ExtensionComponentAdapter.java:73)
	at com.intellij.openapi.extensions.impl.ExtensionComponentAdapter.getExtension(ExtensionComponentAdapter.java:119)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processAdapters(ExtensionPointImpl.java:241)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.getExtensions(ExtensionPointImpl.java:190)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.getContributors(ScriptDefinitionsManager.kt:80)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.reloadScriptDefinitions(ScriptDefinitionsManager.kt:87)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.currentDefinitions(ScriptDefinitionsManager.kt:64)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.findScriptDefinition(ScriptDefinitionsManager.kt:71)
	at org.jetbrains.kotlin.script.ScriptDefinitionProvider$DefaultImpls.findScriptDefinition(KotlinScriptDefinitionProvider.kt:27)
	at org.jetbrains.kotlin.idea.core.script.ScriptDefinitionsManager.findScriptDefinition(ScriptDefinitionsManager.kt:44)
	at org.jetbrains.kotlin.script.KotlinScriptDefinitionProviderKt.getScriptDefinition(KotlinScriptDefinitionProvider.kt:36)
	at org.jetbrains.kotlin.idea.core.script.dependencies.KotlinScriptResolveScopeProvider.getResolveScope(KotlinScriptResolveScopeProvider.kt:31)
	at com.intellij.psi.impl.file.impl.ResolveScopeManagerImpl.lambda$new$0(ResolveScopeManagerImpl.java:63)
	at com.intellij.util.containers.ConcurrentFactoryMap$5.create(ConcurrentFactoryMap.java:234)
	at com.intellij.util.containers.ConcurrentFactoryMap.get(ConcurrentFactoryMap.java:62)
	at com.intellij.psi.impl.file.impl.ResolveScopeManagerImpl.getResolveScopeFromProviders(ResolveScopeManagerImpl.java:81)
	at com.intellij.psi.impl.file.impl.ResolveScopeManagerImpl.getResolveScope(ResolveScopeManagerImpl.java:143)
	at com.intellij.psi.impl.ResolveScopeManager.getElementResolveScope(ResolveScopeManager.java:51)
	at com.intellij.psi.impl.PsiElementBase.getResolveScope(PsiElementBase.java:184)
	at com.intellij.psi.impl.ElementPresentationUtil.getClassKindImpl(ElementPresentationUtil.java:131)
	at com.intellij.psi.impl.ElementPresentationUtil.lambda$getClassKind$0(ElementPresentationUtil.java:106)
	at com.intellij.psi.impl.PsiCachedValueImpl.doCompute(PsiCachedValueImpl.java:49)
	at com.intellij.util.CachedValueBase.getValueWithLock(CachedValueBase.java:222)
	at com.intellij.psi.impl.PsiCachedValue.getValueWithLock(PsiCachedValue.java:60)
	at com.intellij.psi.impl.PsiCachedValueImpl.getValue(PsiCachedValueImpl.java:38)
	at com.intellij.psi.impl.ElementPresentationUtil.getClassKind(ElementPresentationUtil.java:109)
	at com.intellij.psi.impl.ElementPresentationUtil.getFlags(ElementPresentationUtil.java:52)
	at com.intellij.psi.impl.ElementPresentationUtil.createLayeredIcon(ElementPresentationUtil.java:64)
	at com.intellij.psi.impl.PsiClassImplUtil.lambda$static$0(PsiClassImplUtil.java:261)
	at com.intellij.ui.IconDeferrerImpl.deferImpl(IconDeferrerImpl.java:78)
	at com.intellij.ui.IconDeferrerImpl.defer(IconDeferrerImpl.java:68)
	at com.intellij.psi.impl.PsiClassImplUtil.getClassIcon(PsiClassImplUtil.java:281)
	at com.intellij.psi.impl.PsiClassImplUtil.getClassIcon(PsiClassImplUtil.java:268)
	at com.intellij.psi.impl.source.PsiClassImpl.getElementIcon(PsiClassImpl.java:633)
	at com.intellij.psi.impl.ElementBase.computeIconNow(ElementBase.java:119)
	at com.intellij.psi.impl.ElementBase.lambda$static$0(ElementBase.java:59)
	at com.intellij.ui.DeferredIconImpl.evaluate(DeferredIconImpl.java:278)
	at com.intellij.ui.DeferredIconImpl.lambda$null$0(DeferredIconImpl.java:163)
	at com.intellij.ui.IconDeferrerImpl.evaluateDeferred(IconDeferrerImpl.java:111)
	at com.intellij.ui.DeferredIconImpl.lambda$null$1(DeferredIconImpl.java:163)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1142)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runInReadActionWithWriteActionPriority$0(ProgressIndicatorUtils.java:84)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.lambda$runWithWriteActionPriority$2(ProgressIndicatorUtils.java:140)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:157)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:580)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:525)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:85)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:144)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runWithWriteActionPriority(ProgressIndicatorUtils.java:138)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:84)
	at com.intellij.openapi.progress.util.ProgressIndicatorUtils.runInReadActionWithWriteActionPriority(ProgressIndicatorUtils.java:105)
	at com.intellij.ui.DeferredIconImpl.lambda$paintIcon$4(DeferredIconImpl.java:162)
	at com.intellij.util.concurrency.BoundedTaskExecutor$2.run(BoundedTaskExecutor.java:212)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: org.jetbrains.kotlin.script.ScriptTemplatesProvider PluginClassLoader[LivePlugin, 0.6.1 beta] com.intellij.ide.plugins.cl.PluginClassLoader@7563e062
	at com.intellij.ide.plugins.cl.PluginClassLoader.loadClass(PluginClassLoader.java:63)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 70 more
```

**Comment by krasa** [comment]
Every time.
IntelliJ IDEA 2018.1 EAP 181.2784.17

**Comment by krasa** [comment]
Oh, so it happens on every project reopen.

**Comment by osbald** [comment]
Also happens in IDEA 2017.3.3 with current Kotlin 1.2.20-release-IJ2017.3-1 plugin.

**Comment by kreiger** [comment]
Having the same problem with IDEA 2018.1 EAP.

I see that LivePlugin 0.6.2 beta is out for IDEA 2017.
Could i ask that you release a version of LivePlugin for IDEA 2018.1 as well, please?

**Comment by dkandalov** [comment]
Version 0.6.2 for IJ2017 was to fix something else.

As I understand the ClassNotFoundException is fundamentally caused by kotlin plugin and IJ classloader interaction (https://youtrack.jetbrains.com/issue/KT-22658). But it doesn't look like youtrack issue is going to be fixed anytime soon so I guess I need to come up with some workaround 🤔

**Comment by kreiger** [comment]
@dkandalov Thank you for your efforts! :)

**Comment by dkandalov** [comment]
This was fixed in v0.6.3 and should be working in IntelliJ starting from 2019.1 (and hopefully IJ api is more stable now so things won't break again that easily 🙄).

Sorry it took ages to fix.
